src/internal/m365/onedrive/mock/handlers.go

```diff
@@ -233,11 +233,19 @@ func (m GetsItemPermission) GetItemPermission(
 type RestoreHandler struct {
```

Comment:
I feel like this is increasingly turning into something that would be best suited by generated mocks. We're kind of getting to the point where we want to track which calls are made and return info for some of them. That's something that at least some mock packages support out of the box (iirc the golang mocks package does)

Reply:
mm, admittedly, my personal experience is that generated mocks cause more grief than they save.  Lots of time backlogged in fights with extensive generated output overhauls to comply with small changes.  I'm the wrong person to champion this addition.  And, philosophically, I equate an increase in value from generated mocks with a smell.  It suggests that the interface has grown too large and unwieldy, and is a non-idiomatic implementation.  Can't always avoid that, though.

Reply:
>philosophically, I equate an increase in value from generated mocks with a smell. It suggests that the interface has grown too large and unwieldy, and is a non-idiomatic implementation.

I think these can actually be disjoint. The interface that needs tested here isn't exceptionally large, just we want to validate the data passed to said interface is what we expect it to be. That's where mocks can excel because they allow for patterns like `mock.OnFuncX().Expect(param1, param2)` and can be configured to fail the test if there's unexpected calls to functions or more than the expected number of calls

Reply:
> isn't exceptionally large

heh, a matter of opinion, that.

> That's where mocks can excel

I am, to be fair, fully aware of the functionality provided by generative mocks.  And, also to be fair, I'm quite a fan of the behavior.  My primary disagreement comes from projects where maintenance of those mocks at scale can overtake their granular benefits.  Admittedly, those experiences could have related to poor implementations, and I have a misguided opinion.  Nonetheless, the feeling stands.